### PR TITLE
Added: early stopping

### LINKED ISSUE #95: Early Stopping
- [x] CNN Early Stopping implementieren

### PR DESCRIPTION
**Pull request by florinbarbisch**
Nid so vill zum reviewe!

### CODE REVIEW

**Comment by ghost** [comment]
# RUN: CNN, 2023-05-20 16-05-30

## Hyperparameters

| Path        | Param                               | main                    | workspace   |
|-------------|-------------------------------------|-------------------------|-------------|
| params.yaml | dvclive.data.batch_size             | -                       | 64          |
| params.yaml | dvclive.data.num_workers            | -                       | 0           |
| params.yaml | dvclive.data.output_size            | -                       | 6           |
| params.yaml | dvclive.data.params.columns         | numeric                 | -           |
| params.yaml | dvclive.data.train_val_split        | -                       | 0.8         |
| params.yaml | dvclive.data.type                   | Sklearn_Segments        | NDArray     |
| params.yaml | dvclive.model                       | AdaBoostStumpClassifier | CNN         |
| params.yaml | dvclive.model_hparams.learning_rate | 0.5                     | 0.05        |
| params.yaml | dvclive.model_hparams.momentum      | -                       | 0.9         |
| params.yaml | dvclive.model_hparams.n_estimators  | 200                     | -           |
| params.yaml | dvclive.model_hparams.num_epochs    | -                       | 15          |
| params.yaml | dvclive.model_hparams.random_state  | 42                      | -           |
| params.yaml | dvclive.model_hparams.weight_decay  | -                       | 0.0005      |
| params.yaml | dvclive.seed                        | 42                      | 40          |
| params.yaml | train_test_split_ratio02.type       | measurement             | segment     |

## Metrics

| Path                 | Metric           | main    | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | epoch            | -       | 13          | -        |
| dvclive/metrics.json | step             | -       | 1105        | -        |
| dvclive/metrics.json | test.epoch.acc   | 0.70765 | 0.89617     | 0.18851  |
| dvclive/metrics.json | test.epoch.f1    | 0.66619 | 0.88956     | 0.22337  |
| dvclive/metrics.json | test.step.acc    | -       | 0.89617     | -        |
| dvclive/metrics.json | test.step.f1     | -       | 0.88956     | -        |
| dvclive/metrics.json | train.epoch.acc  | 0.72471 | 0.89731     | 0.17259  |
| dvclive/metrics.json | train.epoch.f1   | 0.68623 | 0.88132     | 0.19509  |
| dvclive/metrics.json | train.epoch.loss | -       | 0.25627     | -        |
| dvclive/metrics.json | train.step.acc   | -       | 0.9375      | -        |
| dvclive/metrics.json | train.step.f1    | -       | 0.92125     | -        |
| dvclive/metrics.json | train.step.loss  | -       | 0.18945     | -        |
| dvclive/metrics.json | val.epoch.acc    | -       | 0.88413     | -        |
| dvclive/metrics.json | val.epoch.f1     | -       | 0.87983     | -        |
| dvclive/metrics.json | val.epoch.loss   | -       | 0.3124      | -        |
| dvclive/metrics.json | val.step.acc     | -       | 0.88413     | -        |
| dvclive/metrics.json | val.step.f1      | -       | 0.87983     | -        |
| dvclive/metrics.json | val.step.loss    | -       | 0.3124      | -        |

## Plots

### Confusion Matrix

Confusion Matrix

**Comment by gabrieltorresgamez** [comment]
seems simple and score doesnt seem to worsen, lgtm

**Review by gabrieltorresgamez — APPROVED** [review]

**Comment by ghost** [comment]
# RUN: CNN, 2023-05-26 07-50-21

## Hyperparameters

| Path   | Param   | main   | workspace   |
|--------|---------|--------|-------------|

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix
